Add VQA datasets, task, models, trainers and evalutors + passes full pipeline test

AntreasAntoniou opened the pull request.

AntreasAntoniou commented:
Any chance you could approve this? It's not very important, but worth integrating for now.

justinengelmann reviewed (approved):
Sorry, didn't realise I was sitting on this. Found no critical bugs in this code

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
updates

AntreasAntoniou committed:
update

AntreasAntoniou committed:
feat: Add VQA evaluation and fix answer token concatenation

This commit adds VQA evaluation to the visual question answering task. Additionally, it fixes a bug in the concatenation of question and answer tokens in the `SimpleVQATransformer` class. The length of the answer tokens is now correctly adjusted to match the length of the question tokens.

AntreasAntoniou committed:
Updates

AntreasAntoniou committed:
Fix model test problems -> all passing

AntreasAntoniou committed:
update vqa metrics

AntreasAntoniou committed:
feat(vqa): add logging of QA data during training

This commit adds logging of question and answer data to a wandb table during
training for visual question answering models. The table is named according to
the training phase and contains columns for the global step, question, ground
truth answer, and predicted answer. This will allow for easier error analysis
and identification of common mistakes made by the model during training.
